src/store/RelayQueryWriter.js

```diff
@@ -84,6 +84,15 @@ class RelayQueryWriter extends RelayQueryVisitor<WriterState> {
     responseData: mixed,
     path: RelayQueryPath
   ): void {
+
+
+    invariant(
+      typeof recordID === 'string',
+      'RelayQueryWriter: Expected id `%s` to be string' +
+      ', instead received a %s',
+      recordID, typeof recordID
+    );
+
```

Comment:
`recordID` is typed here as `DataID` which is a string - can you move this invariant to the point where we read the ID out from the actual payload? This will also help to provide contextual error messages (e.g. that the id on the payload for query foo is a number, or that the id for the deleted field on a mutation is incorrect, etc.). `writeRelayQueryPayload` and `writeRelayUpdatePayload` are the two callers of this class.